adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -353,8 +353,15 @@ public void unBindService(final Context context) {
                 @Override
                 public void run() {
                     if (mBound) {
-                        context.unbindService(BrokerAccountServiceConnection.this);
-                        mBound = false;
+                        try {
+                            context.unbindService(BrokerAccountServiceConnection.this);
+                        } catch (final IllegalArgumentException exception) {
+                            // Catching a runtime exception here but if unbindService throws "Service not registered"
```

Comment:
please add p1 bug# as a TODO to understand issue and fix it correctly. thank you! :)

---
In reply to: [109006864](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/804#discussion_r109006864) [](ancestors = 109006864)